internal/retry/policy.go

```diff
@@ -21,81 +21,83 @@
 package retry
 
 import (
-	"context"
 	"time"
 
 	"go.uber.org/yarpc/api/backoff"
-	"go.uber.org/yarpc/api/transport"
 	ibackoff "go.uber.org/yarpc/internal/backoff"
 )
 
-// PolicyProvider returns a retry policy to use for the request,
-// if no policy is available it should return a default policy.
-type PolicyProvider func(context.Context, *transport.Request) *Policy
+// Policy defines how a retry will be applied.  It contains all the information
+// needed to perform a retry.
+type Policy struct {
+	opts policyOptions
+}
 
-var defaultPolicy = Policy{
+// NewPolicy creates a new retry Policy that can be used in retry middleware.
+func NewPolicy(opts ...PolicyOption) *Policy {
+	policyOpts := defaultPolicyOpts
+	for _, opt := range opts {
+		opt.apply(&policyOpts)
+	}
+	return &Policy{opts: policyOpts}
+}
+
+var defaultPolicyOpts = policyOptions{
 	retries:           0,
 	maxRequestTimeout: time.Second,
 	backoffStrategy:   ibackoff.None,
 }
 
-// Policy defines how a retry will be applied.  It contains all the information
-// needed to preform a retry.
-type Policy struct {
-	// retries is the number of attempts we will retry (after the
-	// initial attempt.
+type policyOptions struct {
+	// retries is the number of attempts we will retry (after the initial
+	// attempt).
 	retries uint
 
-	// maxRequestTimeout is the Timeout we will enforce per request (if this
-	// is more than the context deadline, we'll use the context deadline
-	// instead).
+	// maxRequestTimeout is the Timeout we will enforce per request (if this is
+	// more than the context deadline, we'll use the context deadline instead).
```

Comment:
The latter part is sort of implied but if you really want to mention it, just say,

> maxRequestTimeout is the timeout we will enforce for each outgoing request. This will be clamped down to the context deadline.